web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -45,43 +53,54 @@ public static function create(ContainerInterface $container, array $configuratio
       $plugin_definition,
       $container->get('config.factory'),
       $container->get('logger.factory'),
+      $container->get('database'),
       $container->get('http_client_factory'),
+      $container->get('weather_block.save_cities_for_user'),
+      $container->get('cache.default'),
+      $container->get('current_user'),
     );
   }
 
   /**
    * {@inheritdoc}
    */
   public function build(): array {
-    $user = \Drupal::currentUser();
-    $user_id = $user->id();
+    $user_id = $this->accountProxy->id();
+    $cities = $this->cityService->getSavedCitiesForUser($user_id);
+    $api_key = $this->configFactory->get('block.block.my_awesome_theme_weatherdata')->get('settings.key');
+    $weather_data = [];
 
-    $user_config = $this->configuration['users'][$user_id] ?? [];
-    $cities = $user_config['cities'] ?? [];
-    $api_key = $this->configuration['settings']['key'] ?? '';
+    foreach ($cities as $city) {
+      $cache_id = 'weather_block_data_' . md5($city);
 
-    if (!is_array($cities)) {
-      $cities = [];
-    }
+      if (!$cache = $this->cacheBackend->get($cache_id)) {
+        $api_data = $this->cityService->getDataFromApi($city, $api_key);
 
-    $cache_id = 'weather_block_data_' . $user_id . '_' . md5(serialize($cities));
+        if ($api_data === ['no key']) {
+          return [];
+        }
+        $tags = $this->getCacheTags();
 
-    if (!$cache = \Drupal::cache()->get($cache_id)) {
-      $api_data = $this->getDataFromApi($cities, $api_key);
+        $this->cacheBackend->set($cache_id, $api_data, time() + 3600, $tags);
```

Comment:
60 * 60 can be more clear.
So it can be read as 60 minutes

Reply:
Sorry.